src/main/java/org/elasticsearch/index/query/QueryBuilder.java

```diff
@@ -45,4 +45,10 @@
      * @throws IOException
      */
     Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException;
+
+    /**
+     * Validate the query.
+     * @throws ElasticsearchException if the query is not valid, e.g. fields that are needed to create the lucene query are missing.
+     */
+    void validate() throws ElasticsearchException;
```

Comment:
as for the signature here, we might not just want to throw exception at the first error encountered, but rather collect them all. That would mean returning a list of errors, or a single exception that wraps multiple validation errors as we do in ActionRequest#validate. Thoughts on this?